docket_scrap/spiders/docket_spider.py

```diff
@@ -0,0 +1,213 @@
+                docket["description"] = description
+
+                # extracting date from the description
+                date = self.extract_filing_date(description)
+                docket["date"] = date
+
+                # storing it all in temporary storage
+                temporary_storage = docket
+
+            # if the row has two elements
+            elif len(col) == 2:
+
+                # attempting to extract docket_id out of the first element
+                docket_id = self.extract_docket_id(col[0])
+
+                # if id is present, then the filer is merged with previous id
+                if docket_id:
+                    docket["docket_id"] = docket_id
+
+                    # extracting description from second column
+                    description = col[1]
+                    description = self.extract_description(description)
+                    docket["description"] = description
+
+                    # extracting filer from temporary storage
+                    filer = temporary_storage["filer"]
+                    docket["filer"] = filer
+
+                    # extracting file_url that resides in docket_id anchor tag
+                    file_url = self.extract_anchor(docket_id)
+                    docket["file_url"] = file_url
+
+                    # extracting date from the description
+                    date = self.extract_filing_date(description)
+                    docket["date"] = date
+
+                else:
+                    # if id is not present, then it is merged with previous id
+                    filer = col[0]
+                    filer = self.extract_filer(filer)
+                    docket["filer"] = filer
+
+                    # extracting description from second column
+                    description = col[1]
+                    description = self.extract_description(description)
+                    docket["description"] = description
+
+                    # extracting date from description
+                    date = self.extract_filing_date(description)
+                    docket["date"] = date
+
+                    # copying file_url and id from temporary storage
+                    docket["file_url"] = temporary_storage["file_url"]
+                    docket["docket_id"] = temporary_storage["docket_id"]
+
+            # Since the row length is 1, id and description are merged with previous
+            else:
+
+                # extracting filer from the column
+                filer = col[0]
+                filer = self.extract_filer(filer)
+                docket["filer"] = filer
+
+                # copying previous docket data to current
+                docket["file_url"] = temporary_storage["file_url"]
+                docket["docket_id"] = temporary_storage["docket_id"]
+                docket["description"] = temporary_storage["description"]
+                docket["date"] = temporary_storage["date"]
```

Comment:
These if/else conditional blocks can be improved significantly. Only certain items are changed in if/else. The extraction calls are not varied. Move the operations that will always happen outside the conditionals. Only assign the values in if conditions.